src/apps/profiles/models.py

```diff
@@ -34,3 +58,152 @@ def get_short_name(self):
+        return Task.objects.filter(producer=self.producer, creator_id=self.remote_id)
+
+    @property
+    def solutions(self):
+        from tasks.models import Solution
+        return Solution.objects.filter(producer=self.producer, creator_id=self.remote_id)
+
+
+# WILL BE VERY USEFUL
+class GithubUserInfo(models.Model):
+    # Required Info
+    uid = models.CharField(max_length=30, unique=True)
+
+    # Misc/Avatar/Profile
+    login = models.CharField(max_length=100, null=True, blank=True)  # username
+    avatar_url = models.URLField(max_length=100, null=True, blank=True)
+    gravatar_id = models.CharField(max_length=100, null=True, blank=True)
+    html_url = models.URLField(max_length=100, null=True, blank=True)  # Profile URL
+    name = models.CharField(max_length=100, null=True, blank=True)
+    company = models.CharField(max_length=100, null=True, blank=True)
+    bio = models.TextField(max_length=2000, null=True, blank=True)
+    location = models.CharField(max_length=120, null=True, blank=True)
+    created_at = models.DateTimeField(null=True, blank=True)
+    updated_at = models.DateTimeField(null=True, blank=True)
+
+    # API Info
+    node_id = models.CharField(unique=True, max_length=50, default='')
+    url = models.URLField(max_length=100, null=True, blank=True)  # Base API URL
+    followers_url = models.URLField(max_length=100, null=True, blank=True)
+    following_url = models.URLField(max_length=100, null=True, blank=True)
+    gists_url = models.URLField(max_length=100, null=True, blank=True)
+    starred_url = models.URLField(max_length=100, null=True, blank=True)
+    subscriptions_url = models.URLField(max_length=100, null=True, blank=True)
+    organizations_url = models.URLField(max_length=100, null=True, blank=True)
+    repos_url = models.URLField(max_length=100, null=True, blank=True)
+    events_url = models.URLField(max_length=100, null=True, blank=True)
+    received_events_url = models.URLField(max_length=100, null=True, blank=True)
+
+
+# No use yet
+class DockerUserInfo(models.Model):
+    uid = models.CharField(max_length=30, unique=True)
+
+
+# Not so useful
+class LinkedInUserInfo(models.Model):
+    uid = models.CharField(max_length=30, unique=True)
+    firstName = models.CharField(max_length=50, null=True, blank=True)
+    lastName = models.CharField(max_length=50, null=True, blank=True)
+
+
+class AccountMergeRequest(models.Model):
+    master_account = models.ForeignKey(User, related_name='primary_merge_requests', on_delete=models.CASCADE)
+    secondary_account = models.ForeignKey(User, related_name='secondary_merge_requests', on_delete=models.CASCADE)
+    key = models.UUIDField(default=uuid.uuid4, unique=True)
+
+    created = models.DateTimeField(auto_now_add=True)
+
+    def clean(self, *args, **kwargs):
+        # add custom validation here
+        if self.master_account == self.secondary_account:
+            raise ValidationError("Cannot create a merge request between the same account")
```

Comment:
This enforces is at the Django level instead of only DRF. I think this makes more sense since either way we create the object, we would like this validation applied.